pytorch_lightning/strategies/ddp.py

```diff
@@ -196,6 +196,18 @@ def pre_configure_ddp(self):
             )
             self._ddp_kwargs["find_unused_parameters"] = True
 
+        if self.root_device.type == "hpu":
+            self._static_graph = False
+            static_graph = self._ddp_kwargs.get("static_graph")
+            if static_graph:
+                # when _set_static_graph() is called find_unused_parameters does not have any significance.
+                # Resetting the value of find_unused_parameters to False which is the default value to DDP
+                self._ddp_kwargs["find_unused_parameters"] = False
+                self._static_graph = True
+            if static_graph is not None:
+                # DDP does not accept static_graph as a parameter, hence removing it from the list
+                del self._ddp_kwargs["static_graph"]
```

Comment:
yes it does? as of pytorch 1.11 this should be directly passed on the DDP constructor:  https://pytorch.org/docs/master/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel

Reply:
we will need it for PT <= 1.10.2. Will move it to hpu_parallel